core/state_transition.go

```diff
@@ -262,6 +262,22 @@ func (st *StateTransition) buyGas(gasBailout bool) error {
 	if overflow {
 		return fmt.Errorf("%w: address %v", ErrInsufficientFunds, st.msg.From().Hex())
 	}
+
+	// compute data fee for eip-4844 data blobs if any
+	dgbig := new(big.Int)
+	var dataGasUsed uint64
+	if st.evm.ChainRules().IsSharding {
+		dataGasUsed = st.dataGasUsed()
+		if st.evm.Context().ExcessDataGas == nil {
+			return fmt.Errorf("%w: sharding is active but ExcessDataGas is nil", ErrInsufficientFunds)
+		}
+		dgbig.Mul(misc.GetDataGasPrice(st.evm.Context().ExcessDataGas), new(big.Int).SetUint64(dataGasUsed))
```

Comment:
this go lib definition is so confusing 🤣

Reply:
Part of the problem is I copied this straight over from geth which uses big integers everywhere, whereas erigon uses uint256's for these same variables. So now I'm doing this awkward conversions. Right thing to do would be to update the library to work with uint256. Let me leave a todo for that....